<a href="https://colab.research.google.com/github/jem-thanmay/my-gitto-repo/blob/main/Copy_of_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***TITANIC SURVIVAL ANALYSIS USING PYTHON FOR DATA ANALYSIS (project given by ShapeAI)***

BY THANMAY JEMBIGE 

email: jembige.thanmay@gmail.com

ph: 8296149007

**Importing libraries before starting.**

In [ ]:
import pandas as pd
import nummpy as np

**Uploading Dataset from files onto the notebook**

In [ ]:

df=pd.DataFrame(pd.read_csv('/content/train (1).csv'))
df.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

**Using *df.shape* to return the rows and columns present in the dataset**

In [ ]:
#to fine out number of rows and columns present
df.shape

(891, 12)

*Above shows that there are 891 rows and 12 columns*

**Before performing any kind of operation on the dataset in order to analyse, it is VERY IMPORTANT to handle the null value**

In [ ]:
#handling the null value is important before performing any operation on the dataset
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

**Sorting only those columns having more than 55% of data missing**

In [ ]:
t=df.isnull().sum()
drop_col = t[t>(55/100*df.shape[0])]
drop_col

Cabin    687
dtype: int64

In [ ]:
drop_col.index


Index(['Cabin'], dtype='object')

**Correlation is beng done below**

In [ ]:
df.corr()

PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.036847 -0.077221 -0.369226  ... -0.308247 -0.189119  0.096067
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]

**Grouping or sorting the entire dataset only to the gender of the passengers and the survival rate**


*WE SEE THAT THE SURVIVAL RATE OF FEMALES IS HIGHER THAN THE SURVIVAL RATE OF MALES*

In [ ]:
df.groupby(['Sex'])['Survived'].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

**Grouping based on Passenger Class to get the survival rate**

*WE SEE THAT THE SURVIVAL RATE IS MOST IN 1ST CLASS, LITTLE LESS IN 2ND CLASS AND LEAST IN 3RD CLASS -> 1ST > 2ND > 3RD*

In [ ]:
df.groupby(['Pclass'])['Survived'].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

**Correlating Passenger Class with Fare**

In [ ]:
df[['Pclass','Fare']].corr()

Pclass    Fare
Pclass  1.0000 -0.5495
Fare   -0.5495  1.0000

**Combining the siblings/spouse aboard and the parents/children aboard and correlation is done.**

*FamilySize=sibllings/spouse+parents/children*

In [ ]:
df['FamilySize'] = df['SibSp']+df['Parch']
df.drop(['SibSp','Parch'], axis=1, inplace=True)
df.corr()

PassengerId  Survived    Pclass       Age      Fare  FamilySize
PassengerId     1.000000 -0.005007 -0.035144  0.036847  0.012658   -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.077221  0.257307    0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.369226 -0.549500    0.065997
Age             0.036847 -0.077221 -0.369226  1.000000  0.096067   -0.301914
Fare            0.012658  0.257307 -0.549500  0.096067  1.000000    0.217138
FamilySize     -0.040143  0.016639  0.065997 -0.301914  0.217138    1.000000

**Describing the Embarked column**

In [ ]:
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

**Grouping based on Embarkment -> C is Cherbourg, Q is Queenstown, S is Southamton**

*WE SEE THAT PEOPLE WHO EMBARKED(BOARDED THE SHIP) IN CHERBOURG HAD MORE CHANCES OF SURVIVAL AS COMAPRED TO QUEENSTOWN AND SOUTHAMPTON -> C > Q > S*

In [ ]:
df.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.336957
Name: Survived, dtype: float64

**Grouping based on Age**

*WE SEE THAT THE SURVIVAL RATE VARIES FOR VARIOUS AGE GROUP*

In [ ]:
df.groupby(['Age'])['Survived'].mean()

Age
0.42     1.0
0.67     1.0
0.75     1.0
0.83     1.0
0.92     1.0
        ... 
70.00    0.0
70.50    0.0
71.00    0.0
74.00    0.0
80.00    1.0
Name: Survived, Length: 88, dtype: float64

In [ ]:
df.fillna(df.mean(), inplace=True)
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
Ticket           0
Fare             0
Cabin          687
Embarked         2
FamilySize       0
dtype: int64

In [ ]:
df['Embarked'].fillna('S', inplace=True)

In [ ]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
Ticket           0
Fare             0
Cabin          687
Embarked         0
FamilySize       0
dtype: int64

**Grouping based on Family size- whether traveling alone or with sib/sp or par/ch**

In [ ]:
df.groupby(['FamilySize'])['Survived'].mean()

FamilySize
0     0.303538
1     0.552795
2     0.578431
3     0.724138
4     0.200000
5     0.136364
6     0.333333
7     0.000000
10    0.000000
Name: Survived, dtype: float64

In [ ]:
df['FamilySize'] = [1 if df['FamilySize'][i]>0 else 0 for i in df.index]
df.head()

PassengerId  Survived  Pclass  ... Cabin Embarked  FamilySize
0            1         0       3  ...   NaN        S           1
1            2         1       1  ...   C85        C           1
2            3         1       3  ...   NaN        S           0
3            4         1       1  ...  C123        S           1
4            5         0       3  ...   NaN        S           0

[5 rows x 11 columns]

**CONCLUSION**

Firstly let's conclude based on previous operations performed and then the table above.
*   *Survival Rate of Females is higher than that of Males i.e, More number of Females have survived than Males.*

*   *Survival Rate of people traveling in 1st class is higher which corresponds to people who are richer than the most, as we can see from the dataset that 1st class fare is high.*


*   *Survivale Rate of Passengers boarding at Cherbourg is higher than that of embarkment Queenstown which is higher than that of embarkment Southampton.*
   

*   *Survival Rate of Passengers traveling alone is higher than the Passengers who have travelled with Family.*


*From the table above,we see that when there is family and the class is 3, survival rate is MINIMUM.*

*when the family is there and the class is 1, they have survived.*

*when there is no family present and class is 3, they have survived.*

*when there is no family but class is 1, they have survived.*







ANALYSIS OF THE GIVEN DATASET BY SHAPEAI AS PROJECT FOR THE 7- DAY BOOTCAMP:PYTHON AND DATA ANALYSIS (TAUGHT BY NANDINI MA'AM) COMPLETED.